pkg/tsdb/azuremonitor/util/util.go

```diff
@@ -0,0 +1,64 @@
+package util
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/tsdb/azuremonitor/types"
+)
+
+func getBackendUserFromContext(ctx context.Context) (*backend.User, bool) {
+	if val := ctx.Value(backend.User{}); val != nil {
+		user, ok := val.(*backend.User)
+		return user, ok
+	}
+	return nil, false
+}
+
+func InstrumentQueryDataRequest(ctx context.Context, req *http.Request, dsInfo types.DatasourceInfo, client *http.Client, logger log.Logger, logMessage string) (*http.Response, error) {
+	logParams := []interface{}{
+		"url", req.URL.Host,
+		"method", req.Method,
+		"path", req.URL.Path,
+		"cloud", dsInfo.Cloud,
+		"datasourceID", dsInfo.DatasourceID,
+		"subId", dsInfo.Settings.SubscriptionId,
+		"orgId", dsInfo.OrgID,
+	}
+
+	start := time.Now()
+	req.URL = nil
+	res, err := client.Do(req)
+	elapsed := time.Since(start)
+
+	if err != nil {
+		logParams = append(logParams, "status", "internal error")
+		logParams = append(logParams, "error", err.Error())
+	} else {
+		logParams = append(logParams, "status", res.StatusCode)
+	}
+
+	logParams = append(logParams, "duration", elapsed)
+
+	if dsInfo.Credentials != nil {
+		logParams = append(logParams, "authType", dsInfo.Credentials.AzureAuthType())
+	}
+
+	user, ok := getBackendUserFromContext(ctx)
+	if ok {
+		logParams = append(logParams, "uname", user.Email)
```

Comment:
My only query on this PR is the benefit of using `user.Email` over `user.Login` which is already logged? @wbrowne any thoughts on this?

Reply:
Good point @aangelisc! I think it's probably safest to stick with `user.Login` as I think `Email` can be empty